Doc/library/test.rst

```diff
@@ -1716,7 +1716,7 @@ The :mod:`test.support.warnings_helper` module provides support for warnings tes
 
    In this case all warnings are caught and no errors are raised.
 
-   On entry to the context manager, a :class:`WarningRecorder` instance is
+   On entry to the context manager, a :class:`WarningsRecorder` instance is
    returned. The underlying warnings list from
    :func:`~warnings.catch_warnings` is available via the recorder object's
    :attr:`warnings` attribute.  As a convenience, the attributes of the object
```

Comment:
This is one place where it would be real nice if Sphinx applied some simple semantics to the refs. Something which is an attribute clearly can't be a module. I've qualified the reference. As with the `reset` reference, it is now broken.

```
.../Doc/library/test.rst:1719: WARNING: py:attr reference target not found: WarningsRecorder.warnings
```

Both the `reset` and `warnings` items are fully qualified, so even though the links don't work, the full names are displayed.